api/api.go

```diff
@@ -39,38 +62,12 @@ func cacheFlusher(requests chan int, acks chan error) {
 		<-ticker
 
 		func() {
-			// lock database if needed
-			if requests != nil {
-				requests <- ACQUIREDB
-				err := <-acks
-				if err != nil {
-					return
-				}
-				defer func() {
-					requests <- RELEASEDB
-					<-acks
-				}()
+			// if aptly API runs in -no-lock mode,
+			// caches are flushed when DB is closed anyway, no need
+			// to flush them
+			if requests == nil {
+				flushColections()
 			}
```

Comment:
No need to keep enclosing anonymous function since we don't need a scope anymore.